sdk/core/System.ClientModel/src/ModelReaderWriter/ModelReaderWriter.cs

```diff
@@ -101,7 +111,7 @@ public static BinaryData Write(object model, ModelReaderWriterOptions? options =
 
         options ??= ModelReaderWriterOptions.Json;
 
-        return GetInstance<T>().Create(data, options);
+        return options.GetPersistableInterface(GetInstance<T>()).Create(data, options);
```

Comment:
It looks like this won't call through to the IJsonModel implementation if the model is an IJsonModel, because it explicitly calls `IPersistableModel<T>.Create`.  Should this switch on whether T is IJsonModel or IPersistableModel?